presidio-analyzer/presidio_analyzer/predefined_recognizers/third_party/langextract_recognizer.py

```diff
@@ -0,0 +1,138 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import List
+
+from presidio_analyzer.llm_utils import (
+    check_langextract_available,
+    convert_langextract_to_presidio_results,
+    convert_to_langextract_format,
+    extract_lm_config,
+    get_model_config,
+    get_supported_entities,
+    load_prompt_file,
+    load_yaml_examples,
+    load_yaml_file,
+    render_jinja_template,
+    validate_config_fields,
+)
+from presidio_analyzer.lm_recognizer import LMRecognizer
+
+logger = logging.getLogger("presidio-analyzer")
+
+
+class LangExtractRecognizer(LMRecognizer, ABC):
+    """
+    Base class for LangExtract-based PII recognizers.
+
+    Subclasses implement _call_langextract() for specific LLM providers.
+    """
+
+    def __init__(
+        self,
+        config_path: str,
+        name: str = "LangExtract LLM PII",
+        supported_language: str = "en"
+    ):
+        """Initialize LangExtract recognizer.
+
+        :param config_path: Path to configuration file.
+        :param name: Name of the recognizer (provided by subclass).
+        :param supported_language: Language this recognizer supports (default: "en").
+        """
+        check_langextract_available()
+
+        full_config = load_yaml_file(config_path)
+
+        lm_config = extract_lm_config(full_config)
+        langextract_config = full_config.get("langextract", {})
+
+        supported_entities = get_supported_entities(lm_config, langextract_config)
+
+        if not supported_entities:
+            raise ValueError(
+                "Configuration must contain 'supported_entities' in "
+                "'lm_recognizer' or 'langextract'"
+            )
+
+        validate_config_fields(
+            full_config,
+            [
+                ("langextract",),
+                ("langextract", "model"),
+                ("langextract", "model", "model_id"),
+                ("langextract", "entity_mappings"),
+                ("langextract", "prompt_file"),
+                ("langextract", "examples_file"),
+            ]
+        )
+
+        self.config = langextract_config
+        model_config = get_model_config(
+            full_config, provider_key="langextract"
+        )
+        self.model_id = model_config["model_id"]
+        self.temperature = model_config.get("temperature")
+
+        super().__init__(
+            supported_entities=supported_entities,
+            supported_language=supported_language,
+            name=name,
+            version="1.0.0",
+            model_id=self.model_id,
```

Comment:
nit: passing self.model_id and self.temperature to the parent before initializing them makes no sense. Best case it would pass a None, worst case it would change the default if someone changed the default in the parent's code.